### PR TITLE
Allow struct/dataclass types to be used as default arguments

### PR DESCRIPTION
**Pull request by RobertoRRW**
Signed-off-by: Roberto Ruiz <rruiz@inrule.com>

# TL;DR
Pyflyte execution crashes when a task/worflow that takes a dataclass uses a default argument.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Pyflyte [attempts to convert](https://github.com/flyteorg/flytekit/blob/64a3aaafa9ab7353fa02231c998561915f871755/flytekit/clis/sdk_in_container/run.py#L271) struct type literal arguments for tasks/workflows from json string to its expected type. 

I made it so the conversion isn't tried when it isn't necessary. 

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2555

## Follow-up issue
_NA_

### CODE REVIEW

**Review by pingsutw — DISMISSED** [review]
lgtm, thank you for fixing it

**Review by eapolinario — APPROVED** [review]
